apps/web/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { useState } from 'react';
+import './SearchBar.scss';
+
+export interface Props {
+  /** What happens when we use a searchstring to search for results? */
+  handleSearch: (searchString: string) => void;
+  /**  An optional custom test ID for the component.  */
+  customTestId?: string;
+}
+
+// need to add jsdoc comment
+const SearchBar = ({
+  handleSearch,
+  customTestId = 'searchbar',
+}: Props): React.ReactElement => {
+  /** Tracks the string value within input */
+  const [searchString, setSearchString] = useState<string>('');
+
+  /**
+   * Handles 'Enter' key press & invokes `handleSearch` func
+   *
+   * @param {React.KeyboardEvent} event - The keyboard event that triggered the function.
+   */
+  const handleKeyDown = (event: React.KeyboardEvent): void => {
+    if (event.key === 'Enter') {
+      handleSearch(searchString);
+    }
+  };
+
+  return (
+    <div className="searchbar-container" data-testid="searchbar-container">
+      <input
+        data-testid={customTestId}
+        type="search"
+        placeholder="Search Reddit"
+        value={searchString}
+        onChange={(e) => setSearchString(e.target.value)}
```

Comment:
Are you sure that we want to do this? What if I'm searching for "pie"?

I type "p" and the `onChange` event fires...
Then I type "i" and the `onChange` event fires...

Reply:
Went and looked at the actual input on reddit.com:

![search_bar](https://user-images.githubusercontent.com/12022922/228389243-4b699762-20c4-4991-a617-df0fed0e133c.gif)

So we do want to trigger a callback `onChange` but it isn't the same callback that we want to trigger `onKeyDown`.